docs/src/architecture/domain-model.md

```diff
@@ -0,0 +1,121 @@
+      * Step 3: RTS reads all output values from the **Process Image's** _output area_, and writes all read values to Sensor's and Actuator's
+      * then, next cycle starts with Step 1
+
+* **SoC**-based (**S**ystem **o**n **C**hip) Embedded System:
+  * TODO
+
+* **Microcontroller**-based Embedded System:
+  * TODO
+
+## thin-edge device concept
+
+**thin-edge** facilitates IoT functionality to the device it is running on, as well as to devices that are connected to that device.
+All devices are categorized as below:
+  * the **main-device** is the device thin-edge is running on
+  * all devices connected to the **main-device** are referred as **external child-devices**
+  * each device has a unique **device-id**
+
+## thin-edge data concepts
+
+**thin-edge** provides different devices in a standardized representation to the cloud. Therefore **thin-edge** provides different kinds of data concepts.
+
+  * **Measurements**:
+    * contain numeric data produced by sensors (like temperature readings) or calculated data based on information from the **control application**.
+    * a **measurements** consists of one or more numeric **data points** and optionally meta information as names or units
+  * **Variables**:
+    * contain numeric data provided and used by the device, and can be sent to the device (as e.g. "set points")
+    * a **variable** consists of one or more numeric **data points** and optionally meta information as names or units
+  * **Events**:
+    * contain other real-time information from the sensor network, such as the triggering of a door sensor
+    * an **event** is triggered by a **data point** (e.g. when it changes to a specific value as `0` or `1`), and contains an optional message text.
+  * **Alarms**:
+    * similar to Events, but the user or operator of the system has to take action to resolve the alarm
+    * an **alarm** is triggered by a **data point** (e.g. when it changes to a specific value as `0` or `1`), and contains an optional alarm text.
+
+## thin-edge plugin concept
+
+### Plugin
+
+**thin-edge** realizes high level cloud functionality (e.g. _configuration management_ or _log management_) with **plugins**.
+  * a **plugin** accesses _ressources_ and _services_ of the device, as e.g.
+      * the device's _package manager_ is accessed by _software management_
+      * device's _configuration files_ are accessed by _configuration management_
+  * a **plugin** can be an (external) executable (e.g. as the `c8y_configuration_plugin` for _configuration management_)
+    or a thin-edge built-in software component (e.g. as for _software management_)
+  * usually a **plugin** runs on the **main-device**; thus it can access the _resources_ of the **main-device** directly
+  * to access _ressources_ of an **external child-device** a **plugin** needs another component, referred as **child-device agent**
```

Comment:
```suggestion
  * to access _ressources_ of an **external child-device** a **plugin** might need to interact with another component running on the child device, referred to as **child-device agent**, unless the plugin is running directly on the child-device, interacting directly with thin-edge.
```

Reply:
At that place I like to keep that sentence simpler. That suggestion contains many details that are hard to understand in just one sentence. Furthermore those things are outlined in next section about the "Child-Device Agent" in the next lines.

If you feel something is missing in that "Child-Device Agent" section, feel free suggest a change there.